com.unity.ml-agents/Runtime/Actuators/IDiscreteActionMask.cs

```diff
@@ -0,0 +1,43 @@
+using System.Collections.Generic;
+
+namespace Unity.MLAgents.Actuators
+{
+    /// <summary>
+    /// Interface for writing a mask to disable discrete actions for agents for the next decision.
+    /// </summary>
+    internal interface IDiscreteActionMask
+    {
+        /// <summary>
+        /// Modifies an action mask for discrete control agents.
+        /// </summary>
+        /// <remarks>
+        /// When used, the agent will not be able to perform the actions passed as argument
+        /// at the next decision for the specified action branch. The actionIndices correspond
+        /// to the action options the agent will be unable to perform.
+        ///
+        /// See [Agents - Actions] for more information on masking actions.
+        ///
+        /// [Agents - Actions]: https://github.com/Unity-Technologies/ml-agents/blob/release_2_docs/docs/Learning-Environment-Design-Agents.md#actions
+        /// </remarks>
+        /// <param name="branch">The branch for which the actions will be masked.</param>
+        /// <param name="actionIndices">The indices of the masked actions.</param>
+        void WriteMask(int branch, IEnumerable<int> actionIndices);
+
+        /// <summary>
+        /// Get the current mask for an agent.
+        /// </summary>
+        /// <returns>A mask for the agent. A boolean array of length equal to the total number of
+        /// actions.</returns>
+        bool[] GetMask();
+
+        /// <summary>
+        /// Resets the current mask for an agent.
+        /// </summary>
+        void ResetMask();
+
+        /// <summary>
+        ///
+        /// </summary>
+        int CurrentBranchOffset { get; set; }
```

Comment:
Does this need to be part of the interface? It feels specific to the aggregate implementation (and I don't think it's used anywhere)

Reply:
yeah, I went back and forth on this.  I was wondering if we'd want to be able to switch out implementations in ActuatorManager.  You have convinced me to take it back out.